tutorials/04_get_posts/README.md

```diff
@@ -1,28 +1,109 @@
+1.  [**Handle response**](#Handle-response) - Accept the response in a promise callback, then render the results
+1.  [**Example post object**](#Example-post-object) - An example post object from the response list
+
+#### 1. UI <a name="UI"></a>
+
+The source HTML for our UI can be found in [public/index.html](./public/index.html)
+
+There are three input components to the UI.
+
+*   Filters: where we select one of the five built in filter types.
+
+    `<select id="filters" class="form-control" >...`
+
+*   Tag: where we type in a _single_, arbitrary tag. (The Steem blockchain does not support searching on multiple tags)
+
+    `<input id="tag" class="form-control"/>`
 
-## Filters
+*   Get Posts: It's a button. You click it, and we move on to assembling our post.
+    `<button class="btn btn-primary" onclick="getPosts()">Get Posts</button>`
 
-In Steem there are built-in filters `trending`, `hot`, `new`, `active`, `promoted` etc. which helps us to get list of posts. `client.database.getDiscussions(filter, query)` first argument of this function is one of above filter and library fetches posts from those filters.
+![Step-01-UI.png](images/Step-01-UI.png)
 
-## Query (tag, limit)
+#### 2. Construct query <a name="Construct-query"></a>
 
-Second argument of `getDisccusions` function is query.
+The filter and query are constructed within the async, globally available function `getPosts`
 
-*   You can add a tag to filter the posts that you receive from the server, if this field is empty string it will fetch from full list of posts in that selected filter.
-*   You can also limit the number of results you would like to receive from the query
+The `limit` property you see below limits the total number of posts we'll get back to something
+managable. In this case, five.
 
 ```javascript
-var query = {
-    tag: 'steem', // This tag is used to filter the results by a specific post tag
-    limit: 5, // This limit allows us to limit the number of results returned to 5
+const filter = document.getElementById('filters').value;
+const query = {
+    tag: document.getElementById('tag').value,
+    limit: 5,
 };
 ```
 
-## Query Result
+#### 3. API call <a name="API-call"></a>
 
-The result returned form the service is a `JSON` object with the following properties:
+The api call itself is fairly simple. We use `getDiscussions`.
+The first argument, filter, is a simple string.
+The second argument is our query object.
+Like most of dsteem's api functions, `getDiscussions` returns a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+
+```javascript
+client.database
+    .getDiscussions(filter, query)
+    .th..
+```
+
+#### 4. Handle response <a name="Handle-response"></a>
+
+When the promise returned by `getDiscussions` completes successfully, the function we pass to `.then()`
+iterates over the entries response, and constructs html from it.
+
+```javascript
+...ery)
```

Comment:
Is this intentional?

![image](https://user-images.githubusercontent.com/494368/40758485-691925c2-6442-11e8-8364-826b0af01c1f.png)

Reply:
ellipsis...